pkg/rtc/mediatrackreceiver.go

```diff
@@ -681,11 +681,13 @@ func (t *MediaTrackReceiver) GetQualityForDimension(width, height uint32) liveki
 		})
 	}
 
-	// finds the lowest layer that could satisfy client demands
+	// finds the highest layer with smallest dimensions that still satisfy client demands
```

Comment:
Actually, this is a bit of a mess @lukasIO . Need to rework more stuff I think. `SliceStable` is not going to cut it. It needs to happen before that. The layers are stored in a map. Map ordering is not guaranteed. I think what we need to do is to ensure `providedSizes` slice is sorted by corresponding quality. And then pick the closest layer I think.

Currently, it is assumed that  q, h, f are monotonically increasing in size. So, the sorting works out so that we have quality matching size. But, to be correct, I think we should order by quality and pick the highest (closest) layer.

Reply:
thanks for the review, that all makes sense to me! I'll hold off with further changes from my side. Guess changing the tests also makes more sense once there's a clear path on what the behaviour should be. 
Feel free to continue working on this PR or close it entirely as you see fit!

Reply:
We can continue on this PR Lukas. Just want to ensure that we are not going to be blind sided by some historical knowledge of this code path. Once we have comments from DZ, we can take this to conclusion.